crates/ruff_python_parser/src/lexer.rs

```diff
@@ -1114,6 +1292,108 @@ impl std::fmt::Display for LexicalErrorType {
     }
 }
 
+#[derive(Debug)]
+enum StringQuoteChar {
+    Single,
+    Double,
+}
+
+impl TryFrom<char> for StringQuoteChar {
+    type Error = String;
+
+    fn try_from(c: char) -> Result<Self, Self::Error> {
+        match c {
+            '\'' => Ok(Self::Single),
+            '"' => Ok(Self::Double),
+            _ => Err(format!("Invalid string quote character: {c}")),
+        }
+    }
+}
+
+impl StringQuoteChar {
+    const fn as_char(&self) -> char {
+        match self {
+            Self::Single => '\'',
+            Self::Double => '"',
+        }
+    }
+}
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+enum StringQuoteSize {
```

Comment:
Nit: Maybe `Kind`

Reply:
There's a similar implementation for the string quote and size in the `tokenizer.rs` file where `SimpleTokenizer` is present. But, we can't really depend on `ruff_python_trivia`. Currently, I've added an additional implementation here but maybe it could be possible to move it to a common place like `ruff_python_ast`?